include/openssl/e_os2.h

```diff
@@ -206,8 +206,12 @@ extern "C" {
 #  endif
 # endif
 
-# if defined(OPENSSL_SYS_UEFI) && !defined(ssize_t)
-#  define ossl_ssize_t int
+# if defined(OPENSSL_SYS_UEFI) && !defined(ossl_ssize_t)
+#  if !defined(ssize_t)
+#   define ossl_ssize_t int
+#  else
+#   define ossl_ssize_t ssize_t
+#  endif
 #  define OSSL_SSIZE_MAX INT_MAX
```

Comment:
Clarify one thing to me. As far as I understand UEFI is kind of -nostdinc environment with own "standard" headers that don't seem to follow C standard that closely. [This is despite that fact that, judging from commit history, compiler seems to claim quite recent `__STDC_VERSION__` compliance.] Based on above remark you also seem to be quite lax to inconsistencies like "ssize_t is 32- or 64-bit, but it's ok if corresponding max value is 32 bits." Is it correct perception? If so, how can you be sure that it works out in *all* possible and future situations? I mean you might say "it works out [now] [or it at least seems to]", but question is does it have to be limited by statement of *current* state of affairs? And question even is why do *we* have to accept it? And we don't. And we shouldn't. Nobody should. Any piece of code should be readable by itself and consistent with itself, without reader having to make somewhat wild assumptions about platform- or project-specific quirks.

There is one point that might need clarification in the context. I referred to C standard, but ssize_t is not part of C standard, it's POSIX thing. So it's kind of formally incorrect of me to build argument about ssize_t with reference to C standard. But final conclusion is still valid, we are entitled to demand "observable" consistency, and for this reason I still argue against suggested modification in its current form, on grounds that it's potentially inconsistent with itself. I'd say that if UEFI is available in ILP32 and LP64 flavours, then why not simply stick to `long`? All arguments about it being resource-restricted belong elsewhere. For example, if it's so important, why not stick to x32 ABI, i.e. 64-bit calling convention in ILP32 environment?